jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -1005,4 +996,107 @@ public static URL toURL(File file) throws MalformedURLException
     {
         return file.toURI().toURL();
     }
+
+    /**
+     * Factory to create new Resource instance from a reference.
+     */
+    public interface Factory
+    {
+        /**
+         * Create a new Resource from the factory's point of view.
+         * <p>
+         * This is different then {@link ResourceFactory} in that
+         * it must return a {@link Resource} or throw an IOException,
+         * never null.
+         * </p>
+         *
+         * @param reference the string reference.
+         * @return the Resource instance
+         * @throws IOException if unable to create a Resource reference
+         */
+        Resource newResource(String reference) throws IOException;
+    }
+
+    /**
+     * Parse a delimited String of resource references and
+     * return the List of Resources instances it represents.
+     * <p>
+     * Supports glob references that end in {@code /*} or {@code \*}.
+     * Glob references will only iterate through the level specified and will not traverse
+     * found directories within the glob reference.
+     * </p>
+     *
+     * @param delimitedReferences the comma {@code ,} or semicolon {@code ;} delimited
+     * String of resource references.
+     * @return the list of resources parsed from input string.
+     */
+    public static List<Resource> fromReferences(String delimitedReferences) throws IOException
+    {
+        return fromReferences(delimitedReferences, Resource::newResource);
+    }
+
+    /**
+     * Parse a delimited String of resource references and
+     * return the List of Resources instances it represents.
+     * <p>
+     * Supports glob references that end in {@code /*} or {@code \*}.
+     * Glob references will only iterate through the level specified and will not traverse
+     * found directories within the glob reference.
+     * </p>
+     *
+     * @param delimitedReferences the comma {@code ,} or semicolon {@code ;} delimited
+     * String of resource references.
+     * @param resourceFactory the Resource.Factory used to create new Resource references
+     * @return the list of resources parsed from input string.
+     */
+    public static List<Resource> fromReferences(String delimitedReferences, Resource.Factory resourceFactory) throws IOException
```

Comment:
`fromList` or `fromCSV` would be a better name, as `References` doesn't mean anything to me.

Reply:
`fromList` I'm good with, `fromCSV` I'm against, as this doesn't handle CSV, merely delimited. (I would also accept `fromDelimited`)
None of our existing methods with CSV in the name actually handle CSV. (and that bothers me)

Reply:
This layer has been removed in this PR for `jetty-9.4.x`.
It should show up again in a new PR for `jetty-10.0.x`